src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -52,10 +52,19 @@ def to_matrix(x, dtype=np.double, copy=False):
       return x.copy("C"), True
     else:
       return x, False
+  elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
+    if copy: # Copy the matrix if required.
+      return np.ndarray(x.shape,buffer=x.flatten(),dtype=dtype,order = 'C').copy("C"),True
+    else:
+      return np.ndarray(x.shape,buffer=x.flatten(),dtype=dtype, order = 'C'),False
   else:
-    if isinstance(x, pd.core.series.Series):
-      x = pd.DataFrame(x)
-    return np.array(x, copy=True, dtype=dtype, order='C'), True
+    if isinstance(x, pd.core.series.Series) or isinstance(x, pd.DataFrame):
+      y = x.values
+      if copy: # Copy the matrix if required.
+        return np.ndarray(y.shape,buffer=y.flatten(),dtype=dtype,order = 'C').copy("C"),True
+      else:
+        return np.ndarray(y.shape,buffer=y.flatten(),dtype=dtype, order = 'C'),False
```

Comment:
What about the `else` for this `if` block?  We should probably fall back on the original `return np.array(x, copy=True, dtype=dtype, order='C'), True` that was there.